encodings/fastlanes/src/bitpacking/compute/is_constant.rs

```diff
@@ -17,139 +15,73 @@ impl IsConstantKernel for BitPackedEncoding {
     fn is_constant(
         &self,
         array: &BitPackedArray,
-        _opts: &IsConstantOpts,
+        opts: &IsConstantOpts,
     ) -> VortexResult<Option<bool>> {
         match_each_integer_ptype!(array.ptype(), |$P| {
-            bitpacked_is_constant::<$P, {IS_CONST_LANE_WIDTH / size_of::<$P>()}>(array)
+            bitpacked_is_constant::<$P, {IS_CONST_LANE_WIDTH / size_of::<$P>()}>(array, opts)
         })
-        .map(Some)
     }
 }
 
 register_kernel!(IsConstantKernelAdapter(BitPackedEncoding).lift());
 
 fn bitpacked_is_constant<T: BitPacked, const WIDTH: usize>(
     array: &BitPackedArray,
-) -> VortexResult<bool> {
+    opts: &IsConstantOpts,
+) -> VortexResult<Option<bool>> {
     let mut bit_unpack_iterator = array.unpacked_chunks::<T>();
-    let patches = array
-        .patches()
-        .map(|p| {
-            let values = p.values().to_primitive()?;
-            let indices = p.indices().to_primitive()?;
-            let offset = p.offset();
-            VortexResult::Ok((indices, values, offset))
-        })
-        .transpose()?;
+
+    // Bitpacked arrays with patches are only constant if all values are in patches
+    if let Some(patches) = array.patches() {
+        return if patches.num_patches() == array.len() {
+            is_constant_opts(patches.values(), opts)
+        } else {
+            Ok(Some(false))
```

Comment:
`Ok(None)` ? 

Or do we actually enforce during construction that all patch values must not fit in WIDTH bits (and therefore that this condition is met)? If so, we should still probably add a debug_assert to that effect

Reply:
The `WIDTH` here is an annoying side effect of not having `generic_const_exprs` feature. I went with maximal version where we know right now that our compressor will not put in patches values that fit into the encoding

Reply:
right, but that feels like a thing that could subtly break in future? we should enforce that on the BitPackedArray side, not just on the compressor